virttest/qemu_devices/qcontainer.py

```diff
@@ -4031,9 +4031,25 @@ def hostdev_define_by_params(self, name, params, bus=None):
                 "failover_pair_id": params.get("vm_hostdev_failover_pair_id"),
             }
         )
+        # Use hard code 'iommufd0' here, multiple devices use the same iommufd
+        if params.get("iommufd") == "yes":
+            dev_params["iommufd"] = "iommufd0"
```

Comment:
Good idea, I think I understand what you mean, will try to follow this solution to update the PR, thanks!

Reply:
@PaulYuuu Updated, and updated PR https://github.com/autotest/tp-qemu/pull/4163 to match the new design. Could you review again?
Some explanation:
To support multiple devices with multiple iommufd, we need to map iommufd and hostdev, params.get("vm_hostdev_iommufd") will get multiple devices, so we cannot use it directly, we have to map them in a loop. But we don't know the order of current hostdev in hostdev_define_by_params. And we still need to keep the case: multiple hostdev with the same iommufd. So we need "if else" and "break" in this loop to map them carefully.
1) For multiple hostdev with single iommufd case, we need to attach iommufd to the 2nd, 3rd,... hostdevs, so we need the "else", because the iommufd object has been created once for the 1st hostdev
2) For multiple hostdev with multiple iommufd case, when it comes to the next iommufd for the same hostdev in loop, we should not rewrite the iommufd id, so we need "break". "else" is redundant here, but it will be overwritten when it comes to next hostdev, which we hope next iommufd can be used, so it doesn't matter, we have to keep it for case 1).

Thank you so much.